### PR TITLE
Support ignoring file deletions

### PR DESCRIPTION
**Pull request by johnlk**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This change supports ignoring files which are wholly deleted. This seems to be a common ask. The new config variable is called `ignore_file_deletions` and relies on the files API to determine the status of a file.

This change was inspired by #61 , but with updated patterns and tests.

## How to test
I added unit tests for the new code paths to ensure everything is behaving as expected.

## Link to issues addressed
- [ISSUE-66](https://github.com/CodelyTV/pr-size-labeler/issues/66)

### CODE REVIEW

**Comment by johnlk** [comment]
cc: @OnkarRuikar

File: src/github.sh [hunk 1]
```diff
@@ -19,12 +20,18 @@ github::calculate_total_modifications() {
...
   else
+    # NOTE: this code is not resilient to changes w/ > 100 files as we're not paginating
     local -r body=$(curl -sSL -H "Authorization: token $GITHUB_TOKEN" -H "$GITHUB_API_HEADER" "$GITHUB_API_URL/repos/$GITHUB_REPOSITORY/pulls/$pr_number/files?per_page=100")
 
     for file in $(echo "$body" | jq -r '.[] | @base64'); do
       filename=$(jq::base64 '.filename')
+      status=$(jq::base64 '.status')
       ignore=false
 
+      if [ "$ignore_file_deletions" == "true" ] && [ "$status" == "removed" ]; then
```
Comment (OnkarRuikar) [inline src/github.sh:31]: A tiny perf improvement. If all lines are ignored then no point in continuing the loop if the entire file is deleted.
```suggestion
      if [[ ( "$ignore_file_deletions" == "true" || "$ignore_line_deletions" == "true" ) && "$status" == "removed" ]]; then
```

File: README.md [hunk 2]
```diff
@@ -87,6 +88,7 @@ files_to_ignore: |
 
 - PR Size Labeler considers any line addition, deletion, or modification as a change.
 - A PR will be labeled as 'xl' if it exceeds the amount of changes defined in `l_max_size`.
+- `ignore_file_deletions` is distinct from `ignore_line_deletions` in that it only ignores files which are deleted completely.
```
Comment (OnkarRuikar) [inline README.md:91]:
```suggestion
- `ignore_file_deletions` is distinct from `ignore_line_deletions` in that it only ignores files which are deleted completely. If `ignore_line_deletions` is used then using `ignore_file_deletions` is redundant.
```

File: src/github.sh (hunk 1 above)
Comment (johnlk) [reply src/github.sh:31]: yeah good call, slight improvement

**Comment by JavierCane** [comment]
Hi @johnlk, gooooooood work 🎉 🙌

1. I have taken the freedom to commit some `README.md` changes in the branch instead of making them as a PR comment suggestions because GitHub does not allow me to do so due to implying some non-modified lines 😬

   Feel free to discuss any concern regarding the changes 🙏😇

2. Regarding the inspiration on #61, I completely understand that, given the modifications in the project since that PR it is easier to implement it in a new PR (my fault on not reviewing them on time). However, I would propose to add @rwyland
 as the merge commit co-author in order to give him credit for the contribution 😊

   Taking a look to [his contributions to other public repositories](https://github.com/rwyland/Balanced-Google-Plus/commits/master/), it would be just by adding the following to the merge commit:

   ```
   
   Co-authored-by: Rob W <rlw0788@gmail.com>
   ```

**Review by JavierCane — APPROVED** [review]
Hi @johnlk, gooooooood work 🎉 🙌

1. I have taken the freedom to commit some `README.md` changes in the branch instead of making them as a PR comment suggestions because GitHub does not allow me to do so due to implying some non-modified lines 😬

   Feel free to discuss any concern regarding the changes 🙏😇

2. Regarding the inspiration on #61, I completely understand that, given the modifications in the project since that PR it is easier to implement it in a new PR (my fault on not reviewing them on time). However, I would propose to add @rwyland
 as the merge commit co-author in order to give him credit for the contribution 😊

   Taking a look to [his contributions to other public repositories](https://github.com/rwyland/Balanced-Google-Plus/commits/master/), it would be just by adding the following to the merge commit:

   ```
   
   Co-authored-by: Rob W <rlw0788@gmail.com>
   ```

**Pull request merged**